pkgs/games/prismlauncher/default.nix

```diff
@@ -0,0 +1,91 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  jdk8,
+  jdk,
+  zlib,
+  file,
+  wrapQtAppsHook,
+  xorg,
+  libpulseaudio,
+  qtbase,
+  libGL,
+  quazip,
+  glfw,
+  openal,
+  extra-cmake-modules,
+  tomlplusplus,
+  ghc_filesystem,
+  msaClientID ? "",
+  jdks ? [jdk jdk8],
+}: let
+  libnbtplusplus = fetchFromGitHub {
+    owner = "PrismLauncher";
+    repo = "libnbtplusplus";
+    rev = "2203af7eeb48c45398139b583615134efd8d407f";
+    sha256 = "sha256-TvVOjkUobYJD9itQYueELJX3wmecvEdCbJ0FinW2mL4=";
+  };
+in
+  stdenv.mkDerivation {
+    pname = "prismlauncher";
+    version = "unstable-2022-10-18";
+
+    src = fetchFromGitHub {
+      owner = "PrismLauncher";
+      repo = "PrismLauncher";
+      rev = "243600b75babc636bdd1ac0dc8fd5fa4a2db1a0c";
+      sha256 = "sha256-cDwa6NeZew+MubtRmUViLA8gUaoH3AMxD9PdQcgSgSU=";
+    };
+
+    nativeBuildInputs = [extra-cmake-modules tomlplusplus ghc_filesystem cmake file jdk wrapQtAppsHook];
+    buildInputs = [qtbase zlib quazip];
+
+    cmakeFlags = lib.optionals (msaClientID != "") ["-DLauncher_MSA_CLIENT_ID=${msaClientID}"];
+    dontWrapQtApps = true;
+
+    postUnpack = ''
+      rm -rf source/libraries/libnbtplusplus
+      mkdir source/libraries/libnbtplusplus
+      cp -a ${libnbtplusplus}/* source/libraries/libnbtplusplus
+      chmod -R a+r+w source/libraries/libnbtplusplus
+      chown -R $USER: source/libraries/libnbtplusplus
+      ls source/libraries/libnbtplusplus/*
+    '';
+
+    postInstall = let
+      libpath = with xorg;
+        lib.makeLibraryPath [
+          libX11
+          libXext
+          libXcursor
+          libXrandr
+          libXxf86vm
+          libpulseaudio
+          libGL
+          glfw
+          openal
+          stdenv.cc.cc.lib
+        ];
+    in ''
+      # xorg.xrandr needed for LWJGL [2.9.2, 3) https://github.com/LWJGL/lwjgl/issues/128
+      wrapQtApp $out/bin/prismlauncher \
+        --set LD_LIBRARY_PATH /run/opengl-driver/lib:${libpath} \
```

Comment:
Isn't this done by the nixos module?

Reply:
Think it launches a bunch of more dynamic stuff / downloaded things that won't have the right runpath, so it needs to be in the wrapper?

Reply:
That's also what I know, as the lwjgl jars are downloaded.